covid19risk/permission-portal
# Member saving

## Changes
#### Frontend
Updates member saving to be auto-synced with the database and removes the "Save Settings" button (saving is automatic). Based on [this thread](https://covid-watch.slack.com/archives/C013MKUMZK3/p1590436381204200?thread_ts=1590207876.181600&cid=C013MKUMZK3) in Slack.

#### Cloud Functions
Created a function for keeping Firebase Auth user jwt claims and disabled status in sync with users table entries (`syncAuthUserWithCovidWatchUser`), updated userOnUpdate to call it if any of disabled/isAdmin/isSuperAdmin/organizationID is updated (and modified `onCreate` to make use of it as well).

## To Test
- Run the app locally in dev mode and open the dev Firestore web interface in another tab.
- Sign in as admin@soylentgreen.com and go to Manage Members
- Toggle some user's "Role", and "Status" columns, see that those changes are reflected in the corresponding documents in the `users` table in the Firestore web interface
- Change a user's disabled or isAdmin field from the Firestore web interface, see the changes reflected in their respective row on the Manage Members page

Automated test included for backend.

closes #119
